platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/ConsensusWrapper.java

```diff
@@ -55,7 +55,10 @@ public List<ConsensusRound> addEvent(final EventImpl event, final AddressBook ad
 
         final List<ConsensusRound> rounds = new LinkedList<>();
         for (final Map.Entry<Long, List<EventImpl>> entry : roundEvents.entrySet()) {
-            rounds.add(new ConsensusRound(entry.getValue(), new Generations(consensusSupplier.get())));
+            rounds.add(new ConsensusRound(
+                    entry.getValue(),
+                    new Generations(consensusSupplier.get()),
+                    consensusSupplier.get().getMinGenerationNonAncient()));
```

Comment:
this value is contained within Generations

Reply:
Refactored to the following:

```
        for (final Map.Entry<Long, List<EventImpl>> entry : roundEvents.entrySet()) {
            final Generations generations = new Generations(consensusSupplier.get());
            rounds.add(new ConsensusRound(
                    entry.getValue(),
                    generations,
                    generations.getMinGenerationNonAncient()));
        }

```

Would it be ok to pull `generations` out of the loop like this?

```
        final Generations generations = new Generations(consensusSupplier.get());
        for (final Map.Entry<Long, List<EventImpl>> entry : roundEvents.entrySet()) {
            rounds.add(new ConsensusRound(
                    entry.getValue(),
                    generations,
                    generations.getMinGenerationNonAncient()));
        }

```

Reply:
yes. this, in fact, is a slight but, but it will be fixed by the consensus changes